### PR TITLE
Don't delete Hive and Iceberg schema locations with nonempty subdirectories

### PR DESCRIPTION
**Pull request by jirassimok**
This fixes an error in #10146 and #9767 that caused files in subdirectories to be ignored when determining whether to delete schema locations when dropping Hive and Iceberg schemas.

~~(Any empty subdirectories will still be deleted with the schema, though.)~~

### CODE REVIEW

**Review by losipiuk — APPROVED** [review]

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestCreateDropSchema.java [hunk 1]
```diff
@@ -82,19 +82,20 @@ public void testDropSchemaFilesWithLocationWithExternalFile()
     {
         String schemaName = "schema_with_nonempty_location_" + randomTableSuffix();
         String schemaDir = warehouseDirectory + "/schema-with-nonempty-location/";
+        // Use subdirectory to make sure file check is recursive
+        String subDir = schemaDir + "subdir/";
```
[the remark this answers is not in the document]
Comment (jirassimok) [reply testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestCreateDropSchema.java:86]: Those would have been deleted, but I've now changed that as well and added one more test.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestCreateDropSchema.java [hunk 2]
```diff
@@ -54,31 +54,31 @@ public void testCreateDropSchema()
...
-    public void testDropSchemaWithLocationWithoutExternalFiles()
+    public void testDropSchemaFiles()
     {
-        String schemaName = "schema_with_empty_location_" + randomTableSuffix();
-        String schemaDir = warehouseDirectory + "/schema-with-empty-location/";
+        String schemaName = "schema_without_location_" + randomTableSuffix();
+        String schemaDir = format("%s/%s.db/", warehouseDirectory, schemaName);
 
-        onTrino().executeQuery(format("CREATE SCHEMA %s WITH (location = '%s')", schemaName, schemaDir));
+        onTrino().executeQuery(format("CREATE SCHEMA %s", schemaName));
```
Comment (jirassimok) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestCreateDropSchema.java:62]: These lines aren't actually changed; I just switched the order of this test and the next after renaming them, and they're so similar that git didn't realize.

File: testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestCreateDropSchema.java [hunk 3]
```diff
@@ -82,19 +82,37 @@ public void testDropSchemaFilesWithLocationWithExternalFile()
...
+        assertFileExistence(externalFile, true, "external file exists after creating schema");
         onTrino().executeQuery("DROP SCHEMA " + schemaName);
-        assertFileExistence(schemaDir, true, "schema directory exists after dropping schema");
         assertFileExistence(externalFile, true, "external file exists after dropping schema");
 
-        hdfsClient.delete(externalFile);
+        hdfsClient.delete(schemaDir);
+    }
+
+    @Test // make sure empty directories are noticed as well
```
Comment (losipiuk) [inline testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestCreateDropSchema.java]: nit move the comment to method body.

**Review by losipiuk — APPROVED** [review]

**Comment by losipiuk** [comment]
@jirassimok `error-prone-checks` is failing.

**Pull request merged**